src/Core/Content/Media/Core/Application/RemoteThumbnailLoader.php

```diff
@@ -168,19 +173,28 @@ private function getBaseUrl(): string
         return \rtrim($this->filesystem->publicUrl(''), '/');
     }
 
-    private function getUrl(string $mediaUrl, string $mediaPath, string $width, string $height, ?\DateTimeInterface $mediaUpdatedAt): string
+    private function getUrl(MediaEntity|PartialEntity $mediaEntity, string $mediaUrl, string $width, string $height): ?string
     {
         return $this->extensions->publish(
             name: ResolveRemoteThumbnailUrlExtension::NAME,
             extension: new ResolveRemoteThumbnailUrlExtension(
                 $mediaUrl,
-                $mediaPath,
+                $mediaEntity->get('path'),
                 $width,
                 $height,
                 $this->pattern,
-                $mediaUpdatedAt
+                $mediaEntity->get('updatedAt') ?? $mediaEntity->get('createdAt'),
+                $mediaEntity,
             ),
-            function: function (string $mediaUrl, string $mediaPath, string $width, string $height, string $pattern, ?\DateTimeInterface $mediaUpdatedAt) {
+            function: function (
+                string $mediaUrl,
+                string $mediaPath,
+                string $width,
+                string $height,
+                string $pattern,
+                ?\DateTimeInterface $mediaUpdatedAt,
+                Entity $mediaEntity,
+            ): string {
                 $replacements = [
                     str_starts_with($mediaPath, 'http') ? '' : $mediaUrl,
```

Comment:
as we want to remove `mediaPath` and `mediaUpdatedAt` from the extension, we would need to consider the values from the `mediaEntity` right? :thinking: